src/dealer-details.js

```diff
@@ -21,15 +20,15 @@ const createDealerDirectionsURL = (dealer) => {
   return `https://www.google.com/maps/dir/?api=1&destination=${dealerLocationQuery}`;
 };
 
-const dealerConversionEvent = (dealer, action) => {
-  ReactGA.event({
+const dealerConversionEvent = (dealer, action, trackEvent) => {
+  trackEvent({
     category: "Dealer Locator",
     action: action,
     label: _get(dealer, "name"),
```

Comment:
This seems strange to me. The ReactGA API isn't necessarily the same as any given tracking library, but this PR treats any function we pass into it as `ReactGA.event` flavored.

Reply:
That's true, we could pass a more generic event object up and format it for GA in the main project, and labeling category "Dealer Locator" is totally useless.

I think we do need to separate out the action type & a descriptive label, `{action: foo, label: bar}` uses the GA convention but I can't really think of any better way to name the object keys. Any other proposals?

Reply:
I'd say a prop for each of the on-click events that we think should be able to execute some kind of callback. `onDealerWebsiteButtonClick`, `onDealerPhoneClick`, etc. That way we can pass in whatever we feel like (`ReactGA.event({ action: "foo", label: "bar" })` or `GTM.doSomething("foo", "bar")` or `myCoolCallback(42)`) without having any implicit dependencies (the API surface) and we can chain whatever we want, or not.

Reply:
> I'd say a prop for each of the on-click events that we think should be able to execute some kind of callback. `onDealerWebsiteButtonClick`, `onDealerPhoneClick`, etc. That way we can pass in whatever we feel like (`ReactGA.event({ action: "foo", label: "bar" })` or `GTM.doSomething("foo", "bar")` or `myCoolCallback(42)`) without having any implicit dependencies (the API surface) and we can chain whatever we want, or not.

Added & tested with new props for these callbacks! It's a bit more code but should be really flexible